### PR TITLE
Adds a new space ruin

### PR DESCRIPTION
**Pull request by Alberyk**
This pr adds a new cool space ruin made by kyres, replacing the listening outpost and the sol corvette, that were lame. This space ruin has two ghost roles.

This pr also makes so that space ruin sites only open if the research shuttle is moved to an away site, they will be closed otherwise.

### CODE REVIEW

File: code/modules/ghostroles/spawner/human/zenghu.dm [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/datum/ghostspawner/human/zenghu_survivor
+	short_name = "zenghu"
+	name = "Zeng-Hu Survivor"
+	desc = "Survive whatever might lurk in the Zeng-Hu instalation."
```
Comment (fernerr) [inline code/modules/ghostroles/spawner/human/zenghu.dm]:
```suggestion
	desc = "Survive whatever might lurk in the Zeng-Hu installation."
```

File: code/modules/mob/living/simple_animal/hostile/spider_queen.dm [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	melee_damage_lower = 25
+	melee_damage_upper = 30
+	heat_damage_per_tick = 20
+	cold_damage_per_tick = 20
+	faction = "spiders"
+
+	mob_swap_flags = HUMAN|SIMPLE_ANIMAL|SLIME|MONKEY
+	mob_push_flags = ALLMOBS
+
+	attacktext = "bit"
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/spider_queen.dm]:
```suggestion
	attacktext = "bites"
```

File: code/modules/mob/living/simple_animal/hostile/spider_queen.dm [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	max_targets = 1
+
+	invocation_type = SpI_EMOTE
+	invocation = "climbs on the ceiling."
+
+	hud_state = "spider_climb"
+
+/spell/targeted/ceiling_climb/cast(mob/target,var/mob/living/carbon/human/user as mob)
+	..()
+	if(istype(user, /mob/living/simple_animal/hostile/spider_queen))
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/spider_queen.dm]: Maybe check if the mob isn't hovering.

File: code/modules/mob/living/simple_animal/hostile/spider_queen.dm [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	spell_flags = INCLUDEUSER
+	charge_max = 1200 //2 minute
+	max_targets = 1
+
+	invocation_type = SpI_EMOTE
+	invocation = "climbs on the ceiling."
+
+	hud_state = "spider_climb"
+
+/spell/targeted/ceiling_climb/cast(mob/target,var/mob/living/carbon/human/user as mob)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/spider_queen.dm]: This is taking a human as variable. 🤔

Comment (Alberyk) [reply code/modules/mob/living/simple_animal/hostile/spider_queen.dm]: it just works.

File: code/modules/mob/living/simple_animal/hostile/spider_queen.dm (hunk 2 above)
Comment (Alberyk) [reply code/modules/mob/living/simple_animal/hostile/spider_queen.dm]: image
Bit is correct here.

**Pull request closed without merging**